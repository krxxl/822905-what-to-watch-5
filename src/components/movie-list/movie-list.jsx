import React from 'react';
import MovieListItem from '../movie-list-item/movie-list-item';
import PropTypes from 'prop-types';
import withVideo from '../../hocs/with-video/with-video';

const Components = withVideo(MovieListItem);

const MovieList = (props) => {
  const {COUNTFILM, onSmallCardClick} = props;
  let {films} = props;
  films = films.slice(0, COUNTFILM);
  const elements = films.map((film) => {
    const {name, id, previewImage, previewVideoLink} = film;

    return <Components key={id} video={previewVideoLink} name={name} id={id} preview={previewImage} onSmallCardClick={onSmallCardClick} />;
  });
  return (
    <div className="catalog__movies-list">
      {elements}
    </div>
  );  
}

MovieList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    runTime: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.array.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  onSmallCardClick: PropTypes.func.isRequired,
  COUNTFILM: PropTypes.number,
};

export default MovieList;
