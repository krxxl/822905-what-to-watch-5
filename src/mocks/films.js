export default [{
  name: `The Grand Budapest Hotel`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 1,
},
{
  name: `bohemian-rhapsody`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/bohemian-rhapsody.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `2h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 2,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 3,
}, {
  name: `fantastic-beasts-the-crimes-of-grindelwald`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 15m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 4,
}, {
  name: `johnny-english`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/johnny-english.jpg`,
  preview: `img/johnny-english.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 5,
}, {
  name: `macbeth`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/macbeth.jpg`,
  preview: `img/macbeth.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 6,
}, {
  name: `midnight-special`,
  genre: `kids & family`,
  year: `2010`,
  poster: `img/midnight-special.jpg`,
  preview: `img/midnight-special.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 7,
}, {
  name: `moonrise-kingdom`,
  genre: `thriller`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 8,
}, {
  name: `moonrise-kingdom`,
  genre: `horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 9,
}, {
  name: `moonrise-kingdom`,
  genre: `documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 10,
}, {
  name: `moonrise-kingdom`,
  genre: `drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 11,
}, {
  name: `moonrise-kingdom`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 12,
}, {
  name: `moonrise-kingdom`,
  genre: `Romance`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 13,
}, {
  name: `moonrise-kingdom`,
  genre: `Kids & Family`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 14,
}, {
  name: `moonrise-kingdom`,
  genre: `Horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 15,
}, {
  name: `moonrise-kingdom`,
  genre: `Drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 16,
}, {
  name: `moonrise-kingdom`,
  genre: `Documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 17,
}, {
  name: `moonrise-kingdom`,
  genre: `Crime`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 18,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 19,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 20,
},
];
