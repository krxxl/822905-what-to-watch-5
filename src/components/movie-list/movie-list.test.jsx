import React from 'react';
import renderer from 'react-test-renderer';
import {MovieList} from './movie-list';
import films from '../../mocks/films';
import {Router} from 'react-router-dom';
import browserHistory from '../../browser-history';

describe(`Render MovieList`, () => {
  it(`Render MovieList`, () => {
    const tree = renderer
      .create(
          <Router history={browserHistory}>
            <MovieList COUNTFILM={8} isLoading={true} films={films} onSmallCardClick={() => {}} />
          </Router>,
          {
            createNodeMock: () => {
              return {};
            },
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Render MovieList empty`, () => {
    const tree = renderer
      .create(
          <Router history={browserHistory}>
            <MovieList COUNTFILM={8} isLoading={false} films={[]} onSmallCardClick={() => {}} />
          </Router>,
          {
            createNodeMock: () => {
              return {};
            },
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
