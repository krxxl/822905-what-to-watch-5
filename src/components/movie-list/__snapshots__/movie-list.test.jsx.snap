// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieList Render MovieList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/3"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/4"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/5"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/6"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/7"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/bohemian-rhapsody.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/8"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
</div>
`;

exports[`Render MovieList Render MovieList empty 1`] = `
<div
  className="catalog__movies-list"
/>
`;
